### PR TITLE
Add C/C++ test using cppcheck

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Run cppcheck on C/C++ files and update the version if clang-format.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
Tests passes after disabling the `object_tracking_2d` tests that are currently failing on master.

**Comment by stefaniapedrazzi** [comment]
I would temporary disable the `object_tracking_2d` as it is breaking the tests on all the PRs.

File: tests/test_cppcheck.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    for sourceDir in sourceDirs:
+                        if line.startswith(sourceDir):
+                            shouldSkip = False
+                            for skipped in skippedDirs + skippedfiles:
+                                if line.startswith(skipped):
+                                    shouldSkip = True
+                                    break
+                            if not shouldSkip:
+                                command += ' \"' + line + '\"'
+                            continue
```
Comment (vniclas) [inline tests/test_cppcheck.py:77]: This `continue` does not have any effect.

**Review by vniclas — APPROVED** [review]

**Review by vniclas — CHANGES_REQUESTED** [review]
There are errors reported by `cppcheck` but the tests are not failing.

**Comment by stefaniapedrazzi** [comment]
> There are errors reported by `cppcheck` but the tests are not failing.

For the moment we don't have any C/C++ files to check and this is what `cppcheck` is reporting .
I checked and errors in C/C++ makes the test failing correctly.
Moreover as soon as at least one C/C++ file is found the current errors disappear.
So it is correct that the current errors, even if printed in the log, don't make the test fail.

**Comment by vniclas** [comment]
Thanks for following up on this!

**Review by vniclas — APPROVED** [review]

**Review by vniclas — APPROVED** [review]

**Review by passalis — APPROVED** [review]
Thank you! Seems fine.

**Pull request merged**